Purchases/Purchasing/TrialOrIntroPriceEligibilityChecker.swift

```diff
@@ -0,0 +1,152 @@
+                    resultDict[productId] = IntroEligibility(eligibilityStatus: IntroEligibilityStatus.unknown)
+                }
+                for product in products {
+                    guard let sk2ProductDetails = product as? SK2ProductDetails else {
+                        continue
+                    }
+                    // todo: remove when this gets fixed.
+                    // limiting to arm architecture since builds on beta 5 fail if other archs are included
+                    #if arch(arm64)
+                    let sk2Product = sk2ProductDetails.underlyingSK2Product
+                    let maybeIsEligible = await sk2Product.subscription?.isEligibleForIntroOffer
+
+                    let eligibilityStatus: IntroEligibilityStatus
+
+                    if let isEligible = maybeIsEligible {
+                        eligibilityStatus = isEligible ? .eligible : .ineligible
+                    } else {
+                        eligibilityStatus = .unknown
+                    }
+
+                    introDict[sk2ProductDetails.productIdentifier] =
+                        IntroEligibility(eligibilityStatus: eligibilityStatus)
+                    #endif
+                }
+                receiveEligibility(introDict)
+            }
+        }
+    }
+
+    // swiftlint:disable line_length
+    @available(iOS 12.0, macOS 10.14, macCatalyst 13.0, tvOS 12.0, watchOS 6.2, *)
+    private func sk1ModernEligibilityHandler(maybeReceiptData data: Data,
+                                             productIdentifiers: [String],
+                                             completionBlock receiveEligibility: @escaping ReceiveIntroEligibilityBlock) {
+        introEligibilityCalculator
+            .checkTrialOrIntroductoryPriceEligibility(with: data,
+                                                      productIdentifiers: Set(productIdentifiers)) { receivedEligibility, maybeError in
+                if let error = maybeError {
+                    Logger.error(String(format: Strings.receipt.parse_receipt_locally_error,
+                                        error.localizedDescription))
+                    self.backend.getIntroEligibility(appUserID: self.appUserID,
+                                                     receiptData: data,
+                                                     productIdentifiers: productIdentifiers) { result, maybeAnotherError in
+                        if let intoEligibilityError = maybeAnotherError {
+                            Logger.error(String(format: Strings.purchase.unable_to_get_intro_eligibility_with_error,
+                                                intoEligibilityError.localizedDescription))
+                        }
+                        self.operationDispatcher.dispatchOnMainThread {
+                            receiveEligibility(result)
+                        }
+                    }
```

Comment:
maybe we can add an early `return` here so we can remove the `else` and unindent a bit?